taojieTan/assignments/twopointers/lc75/lc75.py

```diff
@@ -0,0 +1,91 @@
+# Problem Statement #
+
+# Given an array containing 0s, 1s and 2s, sort the array in-place. You should treat numbers of the array as objects, hence, we can’t count 0s, 1s, and 2s to recreate the array.
+
+# The flag of the Netherlands consists of three colors: red, white and blue; and since our input array also consists of three different numbers that is why it is called Dutch National Flag problem.
+
+def dutch_flag_sort(arr):
+  # TODO: Write your code here
+
+# input: an unsorted integer array with only 0s, 1s, and 2s
+# output: the same array with sorted integers (ascending order)
+# edge case: empty array => return []
+
+# since we need to sort the array in-place, we are not allowed to use any extra space => not sort method.
+# we can apply two pointers to keep track of leftmost and rightmost indices in the array.
+# we can use while loop to traverse through the array, if the current value is 0, we swap it with the value of leftmost position; 
+# else if it is 2, we swap it with the value at the rightmost position, else if it is 1, we just skip it.
+# this way, we put all 0s at the left, and all 2s at the right, 1 will automatically be at the middle after the loop
```

Comment:
A small tip: code is easier to read if the approach/details are written above/before the function definition.